### PR TITLE
Users/hiyada/use dot net caching

### PR DESCRIPTION
**Pull request by hiyadav**
Cached machineOs as when global.json feature is enabled and multiple versions needs to be downloaded, every time the same os is being detected as the machine remains same. 
This is for faster processing. 

#10885

### CODE REVIEW

File: Tasks/UseDotNetV2/versionfetcher.ts [hunk 1]
```diff
@@ -224,57 +227,61 @@ export class DotNetCoreVersionFetcher {
...
-            throw tl.loc("FailedInDetectingMachineArch", JSON.stringify(ex));
+            catch (ex) {
+                throw tl.loc("FailedInDetectingMachineArch", JSON.stringify(ex));
+            }
+
+            this.machineOsSuffixes = osSuffix;
         }
 
-        return osSuffix;
+        return this.machineOsSuffixes;
```
Comment (bishal-pdMSFT) [inline Tasks/UseDotNetV2/versionfetcher.ts]: you should not return anything now. make it void.

**Comment by bishal-pdMSFT** [comment]
do you want to port it to M157?

**Comment by hiyadav** [comment]
> do you want to port it to M157?

@bishal-pdMSFT not really, as this is only an enhancement. Let me know if you think this should be ported.

**Comment by bishal-pdMSFT** [comment]
> > do you want to port it to M157?
> 
> @bishal-pdMSFT not really, as this is only an enhancement. Let me know if you think this should be ported.

I am fine with not porting

**Pull request merged**